src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -181,20 +181,18 @@ private void EnsureWatchingForExit()
                 {
                     if (!_watchingForExit)
                     {
-                        Debug.Assert(_waitHandle == null);
                         Debug.Assert(_registeredWaitHandle == null);
                         Debug.Assert(Associated, "Process.EnsureWatchingForExit called with no associated process");
                         _watchingForExit = true;
                         try
                         {
-                            _waitHandle = new ProcessWaitHandle(GetWaitState());
-                            _registeredWaitHandle = ThreadPool.RegisterWaitForSingleObject(_waitHandle,
-                                new WaitOrTimerCallback(CompletionCallback), _waitHandle, -1, true);
+                            _completionCallbackContext = new object();
```

Comment:
Can you elaborate on why this new object is needed at all?  What would go wrong if we just had the old check of `if (waitHandleContext != _completionCallbackContext)` without this?

Reply:
The Process instance can be re-used to refer to another OS process (by calling Dispose and Start). Before emitting the Exited event, the context gives us something to check this instance is still the OS process we were tracking when enabling the event.

Reply:
> by calling Dispose and Start

So why not just dispose/null out the wait handle in Dispose and re-create it in Start?  It's fundamentally a different process, and on the Windows side it'd be a different wait handle... wouldn't it make sense to do that on the Unix side as well?

Reply:
Yes, I think that should work and be simpler.